src/Lean/Compiler/IR/EmitLLVM.lean

```diff
@@ -25,9 +25,13 @@ def leanMainFn := "_lean_main"
 
 namespace LLVM
 -- TODO(bollu): instantiate target triple and find out what size_t is.
-def size_tType (llvmctx : LLVM.Context) : IO (LLVM.LLVMType llvmctx) :=
+def size_tType (llvmctx : LLVM.Context) : BaseIO (LLVM.LLVMType llvmctx) :=
   LLVM.i64Type llvmctx
 
+-- TODO(bollu): instantiate target triple and find out what unsigned is.
```

Comment:
We can probably replace this with a static assert that `unsigned` is two bytes because we're definitely not expecting it to ever be something else :)

Reply:
*two* bytes?

Reply:
Uh, that's the minimum size, but we'd rather expect four, yes

Reply:
Relatedly: Can we please remove the usage of wobbly types from lean.h? I see no advantage to using `unsigned` over `uint32_t` and a risk of miscompilations if they ever differ.

Reply:
Strong :+1: on that one from my side.

Reply:
Strong +1 from me as well, this will eliminate all of the wobbliness from C in the backend code.

Reply:
I suggest performing the refactor (either changing it to 4 bytes, or the more aggressive refactor suggested by @digama0) as a subsequent PR.